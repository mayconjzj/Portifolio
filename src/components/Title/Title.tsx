import { twMerge } from 'tailwind-merge';

import { TitleProps } from '@/models/Title';

export const Title = ({ children, ...rest }: TitleProps) => {
  return (
    <>
      <h1
        className={twMerge(
          'text-3xl font-bold relative z-10 after:content-[] after:absolute after:bottom-[-6px] after:left-0 after:w-52 after:h-[2px] after:bg-gradient-to-r after:from-blue-500 after:to-black',
          rest.className
        )}
        {...rest}
      >
        {children}
      </h1>
      ;
    </>
  );
};
