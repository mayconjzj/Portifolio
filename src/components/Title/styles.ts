import tw from 'tailwind-styled-components';

export const Content = tw.h1`
  text-3xl
  font-bold
  relative
  z-10
  after:content-['']
  after:absolute
  after:bottom-[-6px]
  after:left-0 after:w-52
  after:h-[2px]
  after:bg-gradient-to-r
  after:from-blue-500
  after:to-black
`;
