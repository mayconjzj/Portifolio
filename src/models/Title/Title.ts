export type TitleProps = {
  children: React.ReactNode;
  className?: string;
};
