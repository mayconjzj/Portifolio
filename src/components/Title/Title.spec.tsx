import { render } from '@testing-library/react';

import { Title } from './Title';

describe('Title', () => {
  it('should render the title', () => {
    const { container } = render(<Title>Next test</Title>);
    expect(container).toMatchSnapshot();
  });
});
