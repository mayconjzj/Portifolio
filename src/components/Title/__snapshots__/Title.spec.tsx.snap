// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render the title 1`] = `
<div>
  <h1
    class="text-3xl font-bold relative z-10 after:content-[''] after:absolute after:bottom-[-6px] after:left-0 after:w-52 after:h-[2px] after:bg-gradient-to-r after:from-blue-500 after:to-black"
  >
    Next test
  </h1>
</div>
`;
